### PR TITLE
WT-11671 Compact short csuite tests into one task variant

### PR DESCRIPTION
**Pull request by jiechenbo**
The changes done in this pull request aims to clean up the csuite tasks and run them in one task under parallel jobs. I measured the time for the csuite task in the normal and ASAN variant. In the normal variant, the task roughly takes ~30 minutes and for ASAN it takes 90 minutes, to remedy this problem I have made the ASAN variant not running the long running tests and i have updated the long running labels as well, with both the additions i have gotten it down to 24 minutes. 30 minutes is a good target and acceptable for pull request testing.

### CODE REVIEW

**Review by ajmorton — COMMENTED** [review]
Big fan of the reduced line count!
One small issue to deal with regarding runtime of the new test, other than that things are looking good!

File: test/evergreen.yml [hunk 1]
```diff
@@ -627,6 +627,19 @@ functions:
...
+    params:
+      working_dir: "wiredtiger"
+      shell: bash
+      script: |
+        set -o errexit
+        set -o verbose
+        ${PREPARE_TEST_ENV}
+        . test/evergreen/find_cmake.sh
+        cd cmake_build
+        $CTEST -L csuite ${check_args} -j ${num_jobs} --output-on-failure 2>&1
```
Comment (lukech) [inline test/evergreen.yml:641]: "check_args" sounds a bit confusing as it's really used for excluding a given label regex. I saw the same name used in the `make check all` func, which is more reasonable. Suggest renaming to something general e.g. "extra_args" / "exclude_args" (the former is already used in the `make check directory` func.

Comment (jiechenbo) [reply test/evergreen.yml]: I am now directly using make check directory [8a755a8](https://github.com/wiredtiger/wiredtiger/pull/9779/commits/8a755a83a73c544b194a5307752a51ff82272a4d)

File: test/evergreen.yml [hunk 2]
```diff
@@ -627,6 +627,19 @@ functions:
         ${PREPARE_TEST_ENV}
         $(pwd)/test/csuite/${test_name}/test_${test_name} ${test_args|} 2>&1
 
+  "csuite tests all":
```
[the remark this answers is not in the document]
Comment (jiechenbo) [reply test/evergreen.yml]: Hopefully removing wt10461 will reduce the timing down.

File: test/csuite/CMakeLists.txt [hunk 3]
```diff
@@ -100,6 +100,17 @@ define_c_test(
     EXEC_SCRIPT ${CMAKE_CURRENT_SOURCE_DIR}/timestamp_abort/smoke.sh
```
[the remark this answers is not in the document]
Comment (jiechenbo) [reply test/csuite/CMakeLists.txt:101]: done! added it to the label. [8a755a8](https://github.com/wiredtiger/wiredtiger/pull/9779/commits/8a755a83a73c544b194a5307752a51ff82272a4d)

File: test/evergreen/evg_cfg.py [hunk 4]
```diff
@@ -112,14 +115,15 @@ def find_tests_missing_evg_cfg(test_type, dirs, evg_cfg_file):
         else:
             sys.exit("Unsupported test_type '%s'" % test_type)
 
-        # Check if the Evergreen task name exists in current Evergreen configuration
-        if evg_task_name in str(evg_cfg):
+        # Check if the Evergreen task name exists in current Evergreen configuration or in the
+        # csuite CMakeLists file.
+        if evg_task_name in str(evg_cfg) or d in str(cmake_cfg):
```
Comment (lukech) [inline test/evergreen/evg_cfg.py:120]: As reflected by the name, this function aims at finding those missing tests in the Evergreen config file. This added sidecar check with a slightly different purpose of identifying missing tests in the cmake list file - which is less likely to happen. 

It does feel with the merge of individual csuite tasks there's no need to check missing tests for the "csuite" test_type - meaning the relevant code block can be removed. All newly introduced csuite tests in the future (in cmake list file) will be automatically covered by the new combined csuite task.

**Comment by lukech** [comment]
I like the motivation of this PR, and really happy to hear you managed to bring the combined csuite test down to 25 mins. 

Do you have a criteria used for labeling `long_running` tests? It may be worth sharing with the team (and requesting to follow) after the changes get merged. 

Also, I'm not sure if there's consideration around newly added csuite tests in the future. Though each single new test is not "long_running", accumulated together the new combined csuite task is susceptible to crossing the threshold (e.g. 25 mins) over time.

File: test/evergreen/evg_cfg.py (hunk 4 above)
Comment (jiechenbo) [reply test/evergreen/evg_cfg.py:120]: hmm I agree, but it is also possible that we forget to add the csuite test into the CMakeList. @lukech what do you think should be done here?

**Comment by jiechenbo** [comment]
Currently the only long running criteria is based on the task duration being over 10 minutes. I think with newly added csuite tests we should look into creating buckets to alleviate the problem. But i do think that when the time comes we can make this chance.

File: test/evergreen/evg_cfg.py (hunk 4 above)
Comment (lukech) [reply test/evergreen/evg_cfg.py:120]: > it is also possible that we forget to add the csuite test into the CMakeList

It's possible, and the risk was there before this PR. If we really want to introduce the check (i.e. making sure new csuite dirs/tests are added in the cmake list) I'd suggest using a separate ticket/PR and not putting the check in `evg_cfg.py`. It feels like the risk of missing an entry in cmake list is relatively lower than missing a test in Evergreen config file, so I'm too worried about that. 

So my proposal is to remove the newly added cmake list check, and the previous "csuite" test_type checks from `evg_cfg.py`.

**Comment by jiechenbo** [comment]
I have taken a look and the tests are taking less than 25~ minutes. @ajmorton  @lukech please review again and let me know what you think

**Review by ajmorton — APPROVED** [review]
LGTM, thanks!

**Review by lukech — APPROVED** [review]
LGTM

**Comment by jiechenbo** [comment]
evergreen merge

**Pull request merged**